File: Solutions/0205- Isomorphic Strings/Solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        dictionary = dict()

        for i in range(0, len(s)):
            s_char = s[i]
            t_char = t[i]

            if s_char not in dictionary:
                if t_char not in dictionary.values():
                    dictionary[s_char] = t_char
                else:
                    return False
            else:
                if dictionary[s_char] != t_char:
                    return False

        return True

File: Solutions/0205- Isomorphic Strings/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        if (s.length() != t.length())
            return false;

        Map<Character, Character> map1 = new HashMap<>();
        Map<Character, Character> map2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            Character sChar = s.charAt(i);
            Character tChar = t.charAt(i);

            if (!map1.containsKey(sChar) && !map2.containsKey(tChar)) {
                map1.put(sChar,tChar);
                map2.put(tChar, sChar);
            }
            else if (map1.containsKey(sChar) && map2.containsKey(tChar)) {
                if (map1.get(sChar) != tChar || map2.get(tChar) != sChar)
                    return false;
            } else {
                return false;
            }
        }

        return true;

    }
}